### PR TITLE
Add saving of exam scores

### PR DESCRIPTION
**Pull request by delishad21**
Currently, the app does not save the exam scores when it closes. Lets update it so that it saves the scores.

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java
Comment (Zer0Legion) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:1]: great change. would love to see testing with mock values of exam scores!

File: src/main/java/seedu/address/storage/JsonAdaptedExamScore.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package seedu.address.storage;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.exam.Exam;
+import seedu.address.model.person.Score;
+
+/**
+ * Jackson-friendly version of {@link Exam}.
```
Comment (jayllo-c) [inline src/main/java/seedu/address/storage/JsonAdaptedExamScore.java:10]: do you mean json?

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -75,6 +80,11 @@ public JsonAdaptedPerson(Person source) {
         matric = source.getMatric().matricNumber;
         reflection = source.getReflection().reflection;
         studio = source.getStudio().studio;
+        examScores.addAll(source.getScores().entrySet().stream()
```
Comment (jayllo-c) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:83]: great job simplifying the process of collating the list of exam scores!

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -88,6 +98,11 @@ public Person toModelType() throws IllegalValueException {
             personTags.add(tag.toModelType());
         }
 
+        final Map<Exam, Score> personExamScores = new HashMap<>();
+        for (JsonAdaptedExamScore examScore : examScores) {
+            personExamScores.put(examScore.toModelTypeExam(), examScore.toModelTypeScore());
```
Comment (jayllo-c) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:103]: nice job bringing the scores into a map to simplify the subsequent creation of the corresponding person.

**Review by jayllo-c — COMMENTED** [review]
Good job implementing the saving of the exam scores!

File: src/main/java/seedu/address/logic/commands/ExportCommand.java [hunk 4]
```diff
@@ -202,11 +205,26 @@ public CsvSchema buildCsvSchema(JsonNode jsonTree) {
...
-
         CsvMapper csvMapper = new CsvMapper();
+
+        // The Code below removes examscores from the json tree
+        // Potential future implementation: Export Command also exports exam scores
+
+        // Create a new array node to hold the modified json nodes
+        ArrayNode modifiedJsonTree = JsonNodeFactory.instance.arrayNode();
+        // Iterate over each json node in the json tree
+        for (JsonNode jsonNode : jsonTree) {
```
Comment (Pughal77) [inline src/main/java/seedu/address/logic/commands/ExportCommand.java:216]: Perhaps in the future we implement the creation of the JsonTree to include exam scores

**Review by Pughal77 — COMMENTED** [review]
Nice clean implementation of saving of exam scores

**Pull request merged**